programs/ssl/ssl_client2.c

```diff
@@ -1521,6 +1522,13 @@ int main( int argc, char *argv[] )
             goto usage;
     }
 
+    main_buf_len = MAX_REQUEST_SIZE  + 1;
```

Comment:
Suggestion for improvement: could we make this dynamic, based on the `request_size` command-line option? The purpose of this PR is to make our RAM measurements with `scripts/baremetal.h` more meaningful, and this script measures RAM usage on the client. If the client allocates a fixed-size buffer of 20000 bytes, this will dominate the measurements and make them nearly useless. It'd be better to make this buffer only as large as actually needed.

Reply:
Could not use just opt.request_size as it's sometimes -1, 0 , 1 and then tests will fail.
But I did some optimizations to buffer size. After some testing found hopefully smallest possible heap size which works.

Reply:
Nicely done!